### PR TITLE
imagebuildah: honor build output even if build container is not `commited`

### PR DESCRIPTION
**Pull request by flouthoc**
For cases like where Containerfile only contains `FROM alpine` or `FROM
<imagename>` buildah will just perform name tagging and will not
actually commit contents of build container since there is actually no
changes made in the build process. However previously we only used to
export build output when we were actually commiting contents so `-o` or
`--output` was ignored for such edge-cases following commit ensures that
we honor `-o` or `--output` in all scenarios.

Example: Should be functional now
```Dockerfile
FROM alpine
```
```console
buildah build -t test -o type=tar,dest=out.tar .
```

Closes: https://github.com/containers/buildah/issues/4094

#### Does this PR introduce a user-facing change?

```release-note
fix: users can generate build output of image even when contents of build container is not commited
```

### CODE REVIEW

**Comment by flouthoc** [comment]
@rhatdan Rebased @containers/buildah-maintainers @giuseppe @vrothberg PTAL

File: imagebuildah/stage_executor.go [hunk 1]
```diff
@@ -1725,36 +1742,10 @@ func (s *StageExecutor) commit(ctx context.Context, createdBy string, emptyLayer
...
-			// Since modes like setuid,setgid leaves room for executable
-			// to get invoked with different file-system permission its safer
-			// to strip them off for unpriviledged invocation.
-			// See: https://github.com/containers/buildah/pull/3823#discussion_r829376633
-			extractRootfsOpts.StripSetuidBit = true
-			extractRootfsOpts.StripSetgidBit = true
-			extractRootfsOpts.StripXattrs = true
-		}
-		rc, errChan, err := s.builder.ExtractRootfs(options, extractRootfsOpts)
+		err := s.generateBuildOutput(options, buildOutputOption)
```
Comment (rhatdan) [inline imagebuildah/stage_executor.go:1745]: if err := ...

**Review by rhatdan — APPROVED** [review]

File: imagebuildah/stage_executor.go [hunk 2]
```diff
@@ -986,6 +986,23 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
...
+				// and it contains no steps or instructions (i.e Containerfile only contains
+				// `FROM <imagename> and nothing else so we will never end up commiting this
+				// but instead just re-tag image. For such use-cases if `-o` or `--output` was
+				// specified honor that and export the contents of the current build anyways.
+				logrus.Debugf("Generating custom build output with options %q", s.executor.buildOutput)
+				buildOutputOption, err := parse.GetBuildOutput(s.executor.buildOutput)
+				if err != nil {
+					return "", nil, errors.Wrapf(err, "failed to parse build output")
+				}
+				err = s.generateBuildOutput(buildah.CommitOptions{}, buildOutputOption)
```
[the remark this answers is not in the document]
Comment (flouthoc) [reply imagebuildah/stage_executor.go]: fixed

**Comment by flouthoc** [comment]
@rhatdan Could PTAL again.

**Comment by rhatdan** [comment]
LGTM
@giuseppe @vrothberg @umohnani8 PTAL

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold
